### PR TITLE
Cleanup `PpMode` and friends

### PR DESCRIPTION
**Pull request by LeSeulArtichaut**
This PR:
 - Separates `PpSourceMode` and `PpHirMode` to remove invalid states
 - Renames the variant to remove the redundant `Ppm` prefix
 - Adds basic documentation for the different pretty-print modes
 - Cleanups some code to make it more idiomatic
 
Not sure if this is actually useful, but it looks cleaner to me.

### CODE REVIEW

**Comment by LeSeulArtichaut** [comment]
Mhh, maybe r? @jyn514 ?

**Comment by jyn514** [comment]
I'm not the right reviewer.

r? @pnkfelix maybe?

File: compiler/rustc_session/src/config.rs [hunk 1]
```diff
@@ -2130,6 +2111,24 @@ fn parse_pretty(
...
+    };
+
+    if pretty.is_none() {
+        debugging_opts.unpretty.as_ref().map(|a| {
+            // extended with unstable pretty-print variants
+            parse_pretty_inner(efmt, &a, true)
+        })
+    } else {
+        pretty
+    }
```
Comment (LingMan) [inline compiler/rustc_session/src/config.rs:2131]: Maybe `pretty.or_else(...)`?

Comment (LeSeulArtichaut) [reply compiler/rustc_session/src/config.rs:2131]: Uh, I can just return early, actually

**Pull request merged**